devtempstats_client_app/app/src/main/java/devtempstats/client/devtempstatsapp/MainActivity.java

```diff
@@ -0,0 +1,137 @@
+package devtempstats.client.devtempstatsapp;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.os.RemoteException;
+
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
+
+import devtempstats.service.devtempstatsmanager.devtempstatsManager;
+
+
+public class MainActivity extends AppCompatActivity {
+
+    private devtempstatsManager devtempstats = devtempstatsManager.getInstance();
+    private Button cpuButton,cpuMaxButton,cpuAverageButton, gpuButton, gpuMaxButton, gpuAverageButton, ambientButton, ambientMaxButton, ambientAverageButton;
+    private TextView cpuTextView, gpuTextView, ambientTextView;
+    private Float cpuTemp, gpuTemp, ambientTemp;
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        cpuButton = (Button) findViewById(R.id.cpuButton);
+        cpuMaxButton = (Button) findViewById(R.id.cpuMaxButton);
+        cpuAverageButton = (Button) findViewById(R.id.cpuAverageButton);
+        gpuButton = (Button) findViewById(R.id.gpuButton);
+        gpuMaxButton = (Button) findViewById(R.id.gpuMaxButton);
+        gpuAverageButton = (Button) findViewById(R.id.gpuAverageButton);
+        ambientButton = (Button) findViewById(R.id.ambientButton);
+        ambientMaxButton = (Button) findViewById(R.id.ambientMaxButton);
+        ambientAverageButton = (Button) findViewById(R.id.ambientAverageButton);
+        cpuTextView = (TextView) findViewById(R.id.cpuTempTextView);
+        gpuTextView = (TextView) findViewById(R.id.gpuTempTextView);
+        ambientTextView = (TextView) findViewById(R.id.ambientTempTextView);
+
+        cpuButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                try{
+                cpuTemp = devtempstats.getCpuTemperature();
+                cpuTextView.setText("cpu temp is " + cpuTemp.toString());
+                }catch(RemoteException e){
+
```

Comment:
Daca nu stim cum sa tratam o exceptie, cel mai bine este sa o lasam sa fie aruncata, asfel incat, clientul codului nostru sa aiba oportunitatea sa afle de exceptie, si sa o trateze daca stie cum.
Este okay sa facem si "exception swallowing", dar e important sa nu o facem silently. In cazul in care codul arunca  un RemoteException, va fi foarte dificil sa ne dam seama care este problema, fara un log de eroare